### PR TITLE
[PREVIEW] feature/sl 1798/prevent move multisession planner

### PR DESCRIPTION
**Pull request by andrewtaylorkainos**
### Change description ###
Added SnlEventsException if hearing is a multi-session and added a test

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionAction.java [hunk 1]
```diff
@@ -70,6 +71,10 @@ public void getAndValidateEntities() {
             throw new SnlEventsException("Hearing part can not be listed");
         }
 
+        if (hearingPart.getHearing().isMultiSession()) {
+            throw new SnlEventsException("Hearing part cannot be part of a multi-session!");
```
Comment (radoslawlandowskikainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionAction.java]: That message is misleading. I'd say "Cannot assign hearing part which is a part of a multi-session"

File: src/test/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionActionTest.java [hunk 2]
```diff
@@ -130,6 +130,12 @@ public void act_shouldSetClassFieldsToProperState() throws JsonProcessingExcepti
         Mockito.verify(objectMapper, times(2)).writeValueAsString(any());
     }
 
+    @Test(expected = SnlEventsException.class)
+    public void act_shouldThrowServiceException_whenHearingIsMultiSession() {
+        mockedHearingPart.getHearing().setMultiSession(true);
+        action.getAndValidateEntities();
```
Comment (radoslawlandowskikainos) [inline src/test/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionActionTest.java]: There are multiple SnlEventsException possible to be thrown in that method. Please verify the message as well

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionAction.java (hunk 1 above)
Comment (andrewtaylorkainos) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionAction.java]: Done! :)

File: src/test/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionActionTest.java (hunk 2 above)
Comment (andrewtaylorkainos) [reply src/test/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearingpart/AssignHearingPartToSessionActionTest.java]: Done! :)

**Comment by andrewtaylorkainos** [comment]
I agree that it'd be good to have a more refined approach to the 'getAndValidate()' method - which we can hopefully look at soon. 
Although, I think it'd be best, for now, to get the ticket complete without this change so it can be finished before the sprint ends :)

**Pull request merged**